evaluate limits after clipping in Piecewise._eval_interval

smichr opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed

After clipping symbolic limits there may be unevaluated Min/Max functions that will evaluate if they are rebuilt:
```python
>>> from sympy import *
>>> x = var('x')
>>> y = symbols('y', real=True)
>>> g = Piecewise(
...     (1 - x, Interval(0, 1).contains(x)),
...     (1 + x, Interval(-1, 0).contains(x)),
...     (0, True)
...     )
>>>
>>> g.integrate((x,1,y))
...
-Min(1, Max(0, y))**2 + Min(1, Max(y, Min(-1, 0)))**2/2 + Min(1, Max(y, Min(-1, 0))) - 1/2
...
Piecewise((Min(1, Max(-1, y))**2/2 + Min(1, Max(-1, y)) - Min(1, Max(0, y))**2 - 1/2, y < 1), (0, True))
>>>
```
#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* functions
    * doit/simplify should no longer be needed to evaluate limits after Piecewise integration in some cases
<!-- END RELEASE NOTES -->

oscarbenjamin commented:
This could potentially cause slowdowns but I'm not sure how to measure it. Maybe the timings for the slowest tests should be compared with master.

smichr committed:
Piecewise._eval_interval: simplify neg/pos

unevaluated Min/Max can be simplified by just
applying a shallow doit-like simplification.

smichr commented:
> could potentially cause slowdowns

Rough estimate of about 15%. I was able to trace the issue back to the source, however.
